### ISSUE TITLE
Make shadows thrown by DebugGrids configurable

### ISSUE THREAD

**Issue by TheBiochemic** [romenjelly/bevy_debug_grid#4]
Hello there, thanks again for solving the issue with the RenderLayers a while ago!

I found another interesting quirk,
i was trying to essentially get rid of the shadows, that the grid is casting on stuff, so i figured that adding `NotShadowCaster` is sufficient.  The code looked essentially like this
```rust
commands.spawn((
    Grid {
        spacing: 1.0_f32,
        count: 20,
        color: Color::SILVER.with_a(0.75),
        alpha_mode: AlphaMode::Add,
    },
    GridAxis {
        x: Some(Color::RED),
        y: Some(Color::GREEN),
        z: Some(Color::BLUE)
    },
    SubGrid {
        count: 23,
        color: Color::Rgba{red: 0.2, green: 0.2, blue: 0.2, alpha: 0.2},
    },
    TransformBundle::default(),
    VisibilityBundle::default(),
    render_layer,
    NotShadowCaster,
));
```
However, the shadows still show up (even from other layers in 1.12, but that is a known bug, that will be fixed in the next release with https://github.com/bevyengine/bevy/pull/10742).

I built a grid, that corresponds to one pixel size of my textures used, and because the lightsource is at a 45 degree angle, it creates this interesting, but undesired pattern, as seen on the screenshot
![image](https://github.com/romenjelly/bevy_debug_grid/assets/19212761/0d2ffbe0-bfab-41c5-bf2c-71f8b4b4c65a)

Is there a workaround for disabling shadows on the debug grid, or do you have a suggestion on how to approach this?
Thanks in advance :)

**Comment by romenjelly** [comment]
Oh wow, sorry for being two weeks late.
This likely seems to be the same issue as with `RenderLayers`, where `NotShadowCaster` needs to be manually propagated to the spawned children.

I'll try to come up with a general fix in a short while, as adding more `Option<T>` to the query is steadily becoming unmanageable.
Again, sorry for the wait, and thank you for finding this bug!

**Comment by TheBiochemic** [comment]
> I'll try to come up with a general fix in a short while, as adding more `Option<T>` to the query is steadily becoming unmanageable.

Don't worry about it, for now i have temporarily disabled the fine grid :) 
But it would be cool to have it available again.
And yeah i guess the issue is related to the `RenderLayers` Problem.

I would also say it's probably a good idea to have something in Place, that does propagate Components dynamically in some way. It also would have the advantage of being more future proof, since you don't need to deal with potential new Components, that might be added in bevy down the line, maybe build some kind of filter, that iterates through all components of an entity and filters out the ones, that don't need to be put into the children, just a random idea ^^

**Comment by romenjelly** [comment]
Bug has been fixed as of version 0.4.1 7abf315ecb1feb3f13e15f9c126a3737031313a9.
Thanks again for bringing it up!
